File: flask-app/ParseRequestHandler.py
"""

Class to handle parse requests

"""
from antlr4 import *
from LoggingParsers.KerMLv2LoggingParser import KerMLv2LoggingParser
from LoggingParsers.SysMLv2LoggingParser import SysMLv2LoggingParser


class ParseRequestHandler():
    def __init__(self):
        pass

    def handle_request(self, data: dict):
        """Validates input data and runs the parser"""
        if data is None or not isinstance(data, dict):
            return False, "Error: Wrong data format"
        input_string = data.get('input_string')
        if input_string is None:
            return False, "Error: No input string"

        parser_type = data.get('parser_type')
        if parser_type == "0":
            parser = KerMLv2LoggingParser(input_string)
        elif parser_type == "1":
            parser = SysMLv2LoggingParser(input_string)
        else:
            return False, "Error: Wrong parser type. Must be SysMLv2 or KerMLv2"
        parser.parse()
        return True, parser.get_errors()

File: aspnet-app/Services/ParseRequestHandler.cs
using LoggingParsers;
using System;

namespace MyAspNetApp.Services
{
    /// <summary>
    /// Class for validating the input JSON data and running the logging parsers.
    /// </summary>
    public class ParseRequestHandler
    {
        public (bool, string) HandleRequest(dynamic data)
        {
            // Validate input data
            if (data == null || data.input_string == null || data.parser_type == null)
            {
                return (false, "Missing required data.");
            }

            string inputString = data.input_string;

            if (string.IsNullOrWhiteSpace(inputString))
            {
                return (false, "Empty input string.");
            }

            if (!int.TryParse((string)data.parser_type, out int parserType))
            {
                return (false, "Invalid parser type format.");
            }

            LoggingParser parser;

            // Process based on parser type
            if (parserType == 0)
            {
                parser = new KerMLv2LoggingParser(inputString);
            }
            else if (parserType == 1)
            {
                parser = new SysMLv2LoggingParser(inputString);
            }
            else
            {
                return (false, "Invalid parser type. Must be 0 (KerMLv2) or 1 (SysMLv2).");
            }

            // Parse the input string and get errors
            try
            {
                parser.Parse();
                string errors = parser.GetErrors();
                return (true, errors);
            }
            catch (Exception ex)
            {
                return (false, $"Error during parsing: {ex.Message}");
            }
        }
    }
}
